src/Controls/src/Build.Tasks/SetPropertiesVisitor.cs

```diff
@@ -389,13 +389,18 @@ static IEnumerable<Instruction> CompileBindingPath(ElementNode node, ILContext c
 				n = n.Parent as IElementNode;
 			}
 
-			if (dataTypeNode is null)
+			if (dataTypeNode is null) {
+				context.LoggingHelper.LogWarningOrError(10101, context.XamlFilePath, node.LineNumber, node.LinePosition, 0, 0, $"Binding could be compiled if x:DataType is specified", null);
```

Comment:
Should these codes be documented somewhere ? or references from a static that has documentation about these numbers?